.github/workflows/pr.yaml

```diff
@@ -39,6 +41,18 @@ jobs:
     uses: rapidsai/shared-workflows/.github/workflows/conda-cpp-tests.yaml@python-3.12
     with:
       build_type: pull-request
+  conda-java-build:
+    needs: checks
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/conda-java-build.yaml@branch-24.10
+    with:
+      build_type: pull-request
+  conda-java-tests:
+    needs: conda-java-build
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/conda-java-tests.yaml@branch-24.10
```

Comment:
This is the other place we need to use a custom job. [There is no shared-workflows workflow named `conda-java-tests`](https://github.com/rapidsai/shared-workflows/tree/branch-24.12/.github/workflows). Make this look like the code in `test.yaml` except with different parameters (different `build_type:`, no `branch:`, etc.). Look at cuDF for inspiration here.

Also, let's retarget this work to branch-24.12. You'll need to:
- Merge the latest `branch-24.12` into your PR
- Update any lingering references to `branch-24.10` (this line will be one)
- Change the target branch on your PR to `branch-24.12`